### DISCUSSION TITLE
Use minver in github action Docker

### DISCUSSION THREAD

**Discussion by pkuehnel** [adamralph/minver#793] [Q&A] [answered]
I only build my solution with a Docker file within a github action. As the Docker working directory seems to be not inside the git repository minver can not have access to the git tags. Is there any way to create a Docker container with Github Actions and using minver?

Log within the Github Action is:

```
#42 26.65   MinVer: [input] MSBuildProjectDirectory=/src/Projectname/Server
#42 26.66   MinVer: [input] MinVerAutoIncrement=
#42 26.66   MinVer: [input] MinVerBuildMetadata=
#42 26.66   MinVer: [input] MinVerDefaultPreReleasePhase=
#42 26.66   MinVer: [input] MinVerIgnoreHeight=
#42 26.66   MinVer: [input] MinVerMinimumMajorMinor=
#42 26.66   MinVer: [input] MinVerTagPrefix=v
#42 26.66   MinVer: [input] MinVerVerbosity=diagnostic
#42 26.66   MinVer: [input] MinVerVersionOverride=
#42 27.04   MinVer: MinVer 4.1.0+aa8050a19d283115f8b53da0201b0a2af131d2b7.
#42 27.08   MinVer: Running Git: git status --short
#42 27.19   MinVer: Git exit code: 128
#42 27.20   MinVer: Git stdout:
#42 27.20   MinVer: 
#42 27.20   MinVer: Git stderr:
#42 27.20   MinVer: fatal: not a git repository (or any of the parent directories): .git
#42 27.21   MinVer: 
#42 27.29 MinVer : warning MINVER1001: '/src/Projectname/Server' is not a valid Git working directory. Using default version 0.0.0-alpha.0. [/src/Projectname/Server/Projectname.Server.csproj]
#42 27.30   MinVer: The calculated version 0.0.0-alpha.0 satisfies the minimum major minor 0.0.
#42 27.30   MinVer: Calculated version 0.0.0-alpha.0.
#42 27.30   MinVer: [output] MinVerVersion=0.0.0-alpha.0
#42 27.30   MinVer: [output] MinVerMajor=0
#42 27.30   MinVer: [output] MinVerMinor=0
#42 27.30   MinVer: [output] MinVerPatch=0
#42 27.31   MinVer: [output] MinVerPreRelease=alpha.0
#42 27.31   MinVer: [output] MinVerBuildMetadata=
#42 27.31   MinVer: [output] AssemblyVersion=0.0.0.0
#42 27.31   MinVer: [output] FileVersion=0.0.0.0
#42 27.31   MinVer: [output] PackageVersion=0.0.0-alpha.0
#42 27.31   MinVer: [output] Version=0.0.0-alpha.0
```

**Comment by adamralph** [comment]
[SQLStreamStore](https://github.com/SQLStreamStore/SQLStreamStore) also run their build in Docker, so you may want to look at how they are doing it. I'm no Docker expert but it appears they are doing [something explicit](https://github.com/SQLStreamStore/SQLStreamStore/blob/80189cdac3151d5f990600567953ea3be9dad4a2/build.cmd#L5) with the `.git` directory.

**Reply by pkuehnel** [reply]
Hmm... As this results in completly restructuring my pipelines, I am currently trying to play around with the CLI tool and filling an environmentvariable `MINVERBUILDMETADATA` but I don't get the CLI tool to produce the correct format of my version number. My `*.csproj` has these configuration:
```
	<PropertyGroup>
		<MinVerTagPrefix>v</MinVerTagPrefix>
		<MinVerVerbosity>diagnostic</MinVerVerbosity>
	</PropertyGroup>

	<Target Name="MyTarget" AfterTargets="MinVer" >
		<PropertyGroup>
			<FileVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch).$(MinVerPreRelease.Split(`.`)[2]))</FileVersion>
			<PackageVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch).$(MinVerPreRelease.Split(`.`)[2])))</PackageVersion>
			<Version>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch).$(MinVerPreRelease.Split(`.`)[2])))</Version>
		</PropertyGroup>
	</Target>
```

For the CLI tool I am currently this far: `minver --tag-prefix v --verbosity t` but I have no idea how to format my FileVersion, PackageVersion and Version like configured in the `*.csproj` file.


Edit: Btw: `$(MinVerPreRelease.Split(`.`)[2]))` should return the height since last git commit. If you have a better solution for this I would implement that.

**Reply by pkuehnel** [reply]
Ok looks like the environment variable is not passed into the docker build action (or at least minver does notuse it).

**Reply by adamralph** [reply]
@pkuehnel if the `MINVERBUILDMETADATA` env var is present, MinVer _will_ use it.

**Reply by adamralph** [reply]
> For the CLI tool I am currently this far: `minver --tag-prefix v --verbosity t` but I have no idea how to format my FileVersion, PackageVersion and Version like configured in the `*.csproj` file.

See the [FAQ](https://github.com/adamralph/minver#can-i-use-minver-to-version-software-which-is-not-built-using-a-net-sdk-style-project):

> Before building any .NET projects, your build script should run the command-line tool and set the [MINVERVERSIONOVERRIDE](https://github.com/adamralph/minver#options) environment variable to the calculated version. The MinVer package will then use that value rather than calculating the version a second time. This ensures that the command-line tool and the MinVer package produce the same version.

**Reply by adamralph** [reply]
> Edit: Btw: `$(MinVerPreRelease.Split(`.`)[2]))` should return the height since last git commit. If you have a better solution for this I would implement that.

I advise against attempting to do this. Height is just one element of the pre-release identifiers, and it sometimes present (when the current commit is not tagged or is the root commit) and sometimes not (when the current commit is tagged). I advise using the pre-release identifiers in their entirety (e.g. `alpha.0`, `alpha.0.1`, etc.) or not at all, rather than trying to pick out one specific identifier from them (e.g. height).

**Comment by pkuehnel** [comment, marked as answer]
Problem is resolved: [Link to solution](https://stackoverflow.com/a/73472371/8587966)
